javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,256 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj){
+  let answer = [];
+  if(obj.name){
+    let item1 = `<li>name: ${obj.name}</li>`;
+    let item2 = `<li>age: ${obj.age}</li>`;
+    answer.push(item1, item2);
+  }
+  return answer;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  let acc = 0;
+  for (let array of input){
+    for (let number of array){
+      if (number === target){
+        acc += 1;
+      }
+    }
+  }
+  return acc;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let total = 0;
+  for (let array of input){
+    for (let number of array){
+      total += number;
+    }
+  }
+  return total;
+};
+
```

Comment:
Same here, some feedback is this: When you see challenges, always try and think about inputs and outputs. What are the inputs here? An arr of ints.

What do we need to return? A sum. Reduce is always a good higher order function to use when we are tallying something, gathering a count, if we need a unique number etc. 👍  It is quite magical but it can be overwhelming for sure :)